src/client/InteractionClient.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const CommandInteraction = require('../structures/CommandInteraction');
+const { Events, ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+
+let sodium;
+
+function transformCommand(command) {
+  return {
+    name: command.name,
+    description: command.description,
+    options: command.options?.map(function m(o) {
+      return {
+        type: ApplicationCommandOptionType[o.type],
+        name: o.name,
+        description: o.description,
+        default: o.default,
+        required: o.required,
+        choices: o.choices,
+        options: o.options?.map(m),
+      };
+    }),
+  };
+}
+
+/**
+ * Interaction client is used for interactions.
+ *
+ * @example
+ * const client = new InteractionClient({
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * });
+ *
+ * client.on('interactionCreate', () => {
+ *   // automatically handles long responses
+ *   if (will take a long time) {
+ *     doSomethingLong.then((d) => {
+ *       interaction.reply({
+ *         content: 'wow that took long',
+ *       });
+ *     });
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {undefined} client For internal use.
+   */
+  constructor(options, client) {
+    super(options);
+
+    Object.defineProperty(this, 'token', {
+      value: options.token,
+      writable: true,
+    });
+
+    if (client) {
+      this.client = client;
+    } else {
+      this.client = this;
+      this.interactionClient = this;
+
+      Object.defineProperty(this, 'clientID', {
+        value: options.clientID,
+        writable: true,
+      });
+
+      Object.defineProperty(this, 'publicKey', {
+        value: options.publicKey ? Buffer.from(options.publicKey, 'hex') : undefined,
+        writable: true,
+      });
+    }
+  }
+
+  /**
+   * Fetch registered slash commands.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]}
+   */
+  async fetchCommands(guildID) {
+    const clientID = this.client.interactionClient.clientID || (await this.client.fetchApplication()).id;
```

Comment:
```suggestion
    const clientID = this.clientID ?? (await this.client.fetchApplication()).id;
```
not entirely sure why `this.clientID` isn't used here? because thats what it refers to

Reply:
It's because `this.clientID` is only defined if the `InteractionClient` is run standalone.

Reply:
`this.client.interactionClient` will always refer to itself though?